api/cmd/api/setup.go

```diff
@@ -105,55 +104,66 @@ func initImageBuilder(cfg *config.Config) (webserviceBuilder imagebuilder.ImageB
 	}
 
 	var artifactService artifact.Service
-	if cfg.ImageBuilderConfig.ArtifactServiceType == "gcs" {
-		gcsClient, err := gcs.NewClient(context.Background())
-		if err != nil {
-			log.Panicf("%s,failed initializing gcs for mlflow delete package", err.Error())
-		}
-		artifactService = artifact.NewGcsArtifactClient(gcsClient)
-	} else if cfg.ImageBuilderConfig.ArtifactServiceType == "nop" {
+	if cfg.MlflowConfig.ArtifactServiceType == "gcs" {
+		artifactService, err = artifact.NewGcsArtifactClient()
+	} else if cfg.MlflowConfig.ArtifactServiceType == "s3" {
+		artifactService, err = artifact.NewS3ArtifactClient()
+	} else if cfg.MlflowConfig.ArtifactServiceType == "nop" {
 		artifactService = artifact.NewNopArtifactClient()
 	} else {
-		log.Panicf("invalid artifact service type %s", cfg.ImageBuilderConfig.ArtifactServiceType)
+		log.Panicf("invalid artifact service type %s", cfg.MlflowConfig.ArtifactServiceType)
+	}
+
+	if cfg.ImageBuilderConfig.KanikoPushRegistryType != "gcs" &&
+		cfg.ImageBuilderConfig.KanikoPushRegistryType != "docker" {
+		log.Panicf("invalid kaniko push registry type %s", cfg.ImageBuilderConfig.KanikoPushRegistryType)
+	}
+
+	if err != nil {
```

Comment:
very nit: but the err on this line belong to this [if else](https://github.com/caraml-dev/merlin/blob/a63f15f06e2801c9392d965586846307a2d720bf/api/cmd/api/setup.go#L107-L115), so maybe can put it after this

Reply:
Oh do you mean like move this block
```go
if err != nil {
	log.Panicf("%s,failed initializing mlflow artifact service", err.Error())
}
```
to the part just after this
```go
if cfg.MlflowConfig.ArtifactServiceType == "gcs" {
	artifactService, err = artifact.NewGcsArtifactClient()
} else if cfg.MlflowConfig.ArtifactServiceType == "s3" {
	artifactService, err = artifact.NewS3ArtifactClient()
} else if cfg.MlflowConfig.ArtifactServiceType == "nop" {
	artifactService = artifact.NewNopArtifactClient()
} else {
	log.Panicf("invalid artifact service type %s", cfg.MlflowConfig.ArtifactServiceType)
}
```
so the final result looks like this?
```go
if cfg.MlflowConfig.ArtifactServiceType == "gcs" {
	artifactService, err = artifact.NewGcsArtifactClient()
} else if cfg.MlflowConfig.ArtifactServiceType == "s3" {
	artifactService, err = artifact.NewS3ArtifactClient()
} else if cfg.MlflowConfig.ArtifactServiceType == "nop" {
	artifactService = artifact.NewNopArtifactClient()
} else {
	log.Panicf("invalid artifact service type %s", cfg.MlflowConfig.ArtifactServiceType)
}
if err != nil {
	log.Panicf("%s,failed initializing mlflow artifact service", err.Error())
}
```

Reply:
Yess, I'm thinking of moving it like that, because the `err` belongs to that if else codeblock

Reply:
Got it, it should be updated now! Thanks!